serviceconnector/src/validateServiceConnector/ValidateLinkerStep.ts

```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardExecuteStep, nonNullValue } from "@microsoft/vscode-azext-utils";
+import { IPickLinkerContext } from "../deleteServiceConnector/IPickLinkerContext";
+import { createServiceConnectorClient } from "../serviceConnectorClient";
+
+export class ValidateLinkerStep extends AzureWizardExecuteStep<IPickLinkerContext> {
+    public priority: number = 10;
+
+    public async execute(context: IPickLinkerContext): Promise<void> {
+        const client = await createServiceConnectorClient(context.credentials);
+        const response = await client.linker.beginValidateAndWait(nonNullValue(context.sourceResourceUri), nonNullValue(context.linkerName));
+        for (const detail of nonNullValue(response.validationDetail)) {
+            if (detail.result === "failure") {
+                throw new Error(detail.description);
```

Comment:
Something we can deal with later: throwing an error here will tell our telemetry that this wizard actually failed to execute. But in reality the step succeeded in doing it's job, the linker is just invalid.

Reply:
Maybe a `UserCancelledError` with a good cancellation step would be more accurate?